src/tesseract_sinter_compat.pybind.h

```diff
@@ -0,0 +1,332 @@
+          "Input array's second dimension does not match num_detector_bytes.");
+    }
+
+    const size_t num_shots = bit_packed_detection_event_data.shape(0);
+    const uint64_t num_observable_bytes = (num_observables + 7) / 8;
+
+    // Result buffer to store the predicted observables for all shots.
+    auto result_array =
+        py::array_t<uint8_t>({(py::ssize_t)num_shots, (py::ssize_t)num_observable_bytes});
+    auto result_buffer = result_array.mutable_data();
+
+    const uint8_t* detections_data = bit_packed_detection_event_data.data();
+    const size_t detections_stride = bit_packed_detection_event_data.strides(0);
+
+    // Loop through each shot and decode it with TesseractDecoder.
+    for (size_t shot = 0; shot < num_shots; ++shot) {
+      const uint8_t* single_shot_data = detections_data + shot * detections_stride;
+
+      // Unpack the shot data into a vector of indices of fired detectors.
+      std::vector<uint64_t> detections;
+      for (uint64_t i = 0; i < num_detectors; ++i) {
+        if ((single_shot_data[i / 8] >> (i % 8)) & 1) {
+          detections.push_back(i);
+        }
+      }
+
+      // Decode with TesseractDecoder.
+      std::vector<int> predictions = decoder->decode(detections);
+
+      // Store predictions into the output buffer
+      uint8_t* single_result_buffer = result_buffer + shot * num_observable_bytes;
+      std::fill(single_result_buffer, single_result_buffer + num_observable_bytes, 0);
+      for (int obs_index : predictions) {
+        if (obs_index >= 0 && obs_index < num_observables) {
+          single_result_buffer[obs_index / 8] ^= (1 << (obs_index % 8));
+        }
+      }
+    }
+
+    // Return the result.
+    return result_array;
+  }
+};
+
+//--------------------------------------------------------------------------------------------------
+// This struct implements the sinter.Decoder API. It is responsible for creating and compiling
+// a decoder for a specific Detector Error Model (DEM).
+//--------------------------------------------------------------------------------------------------
+struct TesseractSinterDecoder {
+  // Use TesseractConfig as an integrated property.
+  TesseractConfig config;
+
+  // Default constructor
+  TesseractSinterDecoder() : config(TesseractConfig()) {}
+
+  // Constructor with TesseractConfig parameter
+  TesseractSinterDecoder(const TesseractConfig& config_in) : config(config_in) {}
+
+  bool operator==(const TesseractSinterDecoder& other) const {
+    return true;
+  }
+
+  bool operator!=(const TesseractSinterDecoder& other) const {
+    return !(*this == other);
+  }
+
+  // Take a string representation of the DEM, parse the DEM and return a compiled decoder instance.
+  TesseractSinterCompiledDecoder compile_decoder_for_dem(const py::object& dem) {
+    const stim::DetectorErrorModel stim_dem(py::cast<std::string>(py::str(dem)).c_str());
+
+    TesseractConfig local_config = config;
+    local_config.dem = stim_dem;
```

Comment:
It would be better if the first argument of the `TesseractConfig` (at least in the python bindings) was optional and defaulted to None (or some other default).

At the moment, constructing a custom sinter decoder requires adding this dummy DEM, which is a bit ugly:

```
import stim
from tesseract_decoder.tesseract_sinter_compat import TesseractSinterDecoder
from tesseract_decoder.tesseract import TesseractConfig

config = TesseractConfig(dem=stim.DetectorErrorModel(), det_beam=15)
sinter_decoder = TesseractSinterDecoder(config=config)
```

I guess there could then also be a `TesseractConfig.compile_decoder_for_dem(dem: stim.DetectorErrorModel) -> TesseractDecoder` method as well for using the standard decoding class.